spec/controllers/V1/tasks_controller_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+describe V1::TasksController, :type => :controller do
+
+  let!(:project) { create(:project) }
+
+  describe "PUT update_task_state" do
+    let!(:task) { create(:task, project_id: project.id) }
+    let!(:update_params) do { project_id: project.id, task_id: task.id, state: "in-progress" } end
+
+    it "responds with 200" do
+      put 'update_task_state', update_params
+      expect(response.status).to eql(200)
+    end
+
+    it "successfully updates the task" do
+      expect(task.state).to eql("todo")
+
+      put 'update_task_state', update_params
+      body = JSON.parse(response.body)
+
+      task.reload
+      expect(body["state"]).to eql("in-progress")
+      expect(task.state).to eql("in-progress")
+    end
+
+    context "when a task is updated to done" do
+      let!(:update_params) do { project_id: project.id, task_id: task.id, state: "done" } end
+
+      it "sends an sms message when updated to done" do
+        put 'update_task_state', update_params
+        body = JSON.parse(response.body)
+        expect(body).to include("completed")
```

Comment:
I originally wanted to get webmock to work with this, but I was running into a lot of issues trying to set it up and was running out of time. I had also tried setting up a separate sms gem and mocking the Twilio service itself for the tests, but the gem wasn't doing what I wanted and I couldn't get the service to work. If I had longer, I would have tried to keep working through the webmock issues. So I tried checking if the json response came back when the task was updated to done and then I manually tested to make sure my sms messages were working correctly in the rails console.